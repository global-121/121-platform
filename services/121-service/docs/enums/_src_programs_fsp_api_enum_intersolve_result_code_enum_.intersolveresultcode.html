<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>IntersolveResultCode | 121-service</title>
	<meta name="description" content="Documentation for 121-service">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">121-service</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
							<input type="checkbox" id="tsd-filter-only-exported" />
							<label class="tsd-widget" for="tsd-filter-only-exported">Only exported</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../globals.html">Globals</a>
				</li>
				<li>
					<a href="../modules/_src_programs_fsp_api_enum_intersolve_result_code_enum_.html">&quot;src/programs/fsp/api/enum/intersolve-result-code.enum&quot;</a>
				</li>
				<li>
					<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html">IntersolveResultCode</a>
				</li>
			</ul>
			<h1>Enumeration IntersolveResultCode</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Enumeration members</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#authorizationfailed" class="tsd-kind-icon">Authorization<wbr>Failed</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cannotcancel" class="tsd-kind-icon">Cannot<wbr>Cancel</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardblocked" class="tsd-kind-icon">Card<wbr>Blocked</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardexpired" class="tsd-kind-icon">Card<wbr>Expired</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardinactive" class="tsd-kind-icon">Card<wbr>Inactive</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardorcustomerdoesnotexist" class="tsd-kind-icon">Card<wbr>OrCustomer<wbr>Does<wbr>Not<wbr>Exist</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardunavailable" class="tsd-kind-icon">Card<wbr>Unavailable</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#eaninvalid" class="tsd-kind-icon">EANInvalid</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#incorrectactivationorpurchasevalue" class="tsd-kind-icon">Incorrect<wbr>Activation<wbr>OrPurchase<wbr>Value</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#invalidorunknownretailer" class="tsd-kind-icon">Invalid<wbr>OrUnknown<wbr>Retailer</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#invalidparameters" class="tsd-kind-icon">Invalid<wbr>Parameters</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#ok" class="tsd-kind-icon">Ok</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#pininvalid" class="tsd-kind-icon">PINInvalid</a></li>
								<li class="tsd-kind-enum-member tsd-parent-kind-enum"><a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#unabletocancel" class="tsd-kind-icon">Unable<wbr>ToCancel</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Enumeration members</h2>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="authorizationfailed" class="tsd-anchor"></a>
					<h3>Authorization<wbr>Failed</h3>
					<div class="tsd-signature tsd-kind-icon">Authorization<wbr>Failed<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 3</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L3">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:3</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="cannotcancel" class="tsd-anchor"></a>
					<h3>Cannot<wbr>Cancel</h3>
					<div class="tsd-signature tsd-kind-icon">Cannot<wbr>Cancel<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 19</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L7">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:7</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="cardblocked" class="tsd-anchor"></a>
					<h3>Card<wbr>Blocked</h3>
					<div class="tsd-signature tsd-kind-icon">Card<wbr>Blocked<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 103</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L10">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:10</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="cardexpired" class="tsd-anchor"></a>
					<h3>Card<wbr>Expired</h3>
					<div class="tsd-signature tsd-kind-icon">Card<wbr>Expired<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 102</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L9">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:9</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="cardinactive" class="tsd-anchor"></a>
					<h3>Card<wbr>Inactive</h3>
					<div class="tsd-signature tsd-kind-icon">Card<wbr>Inactive<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 119</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L11">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:11</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="cardorcustomerdoesnotexist" class="tsd-anchor"></a>
					<h3>Card<wbr>OrCustomer<wbr>Does<wbr>Not<wbr>Exist</h3>
					<div class="tsd-signature tsd-kind-icon">Card<wbr>OrCustomer<wbr>Does<wbr>Not<wbr>Exist<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 101</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L8">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:8</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="cardunavailable" class="tsd-anchor"></a>
					<h3>Card<wbr>Unavailable</h3>
					<div class="tsd-signature tsd-kind-icon">Card<wbr>Unavailable<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 127</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L13">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:13</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="eaninvalid" class="tsd-anchor"></a>
					<h3>EANInvalid</h3>
					<div class="tsd-signature tsd-kind-icon">EANInvalid<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 212</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L15">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:15</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="incorrectactivationorpurchasevalue" class="tsd-anchor"></a>
					<h3>Incorrect<wbr>Activation<wbr>OrPurchase<wbr>Value</h3>
					<div class="tsd-signature tsd-kind-icon">Incorrect<wbr>Activation<wbr>OrPurchase<wbr>Value<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 120</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L12">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:12</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="invalidorunknownretailer" class="tsd-anchor"></a>
					<h3>Invalid<wbr>OrUnknown<wbr>Retailer</h3>
					<div class="tsd-signature tsd-kind-icon">Invalid<wbr>OrUnknown<wbr>Retailer<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 9</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L5">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:5</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="invalidparameters" class="tsd-anchor"></a>
					<h3>Invalid<wbr>Parameters</h3>
					<div class="tsd-signature tsd-kind-icon">Invalid<wbr>Parameters<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 5</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L4">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:4</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="ok" class="tsd-anchor"></a>
					<h3>Ok</h3>
					<div class="tsd-signature tsd-kind-icon">Ok<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 0</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L2">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:2</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="pininvalid" class="tsd-anchor"></a>
					<h3>PINInvalid</h3>
					<div class="tsd-signature tsd-kind-icon">PINInvalid<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 211</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L14">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:14</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-enum-member tsd-parent-kind-enum">
					<a name="unabletocancel" class="tsd-anchor"></a>
					<h3>Unable<wbr>ToCancel</h3>
					<div class="tsd-signature tsd-kind-icon">Unable<wbr>ToCancel<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol"> = 18</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/diderikvw/121-platform/blob/eab6cf64/services/121-service/src/programs/fsp/api/enum/intersolve-result-code.enum.ts#L6">src/programs/fsp/api/enum/intersolve-result-code.enum.ts:6</a></li>
						</ul>
					</aside>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class="globals  ">
						<a href="../globals.html"><em>Globals</em></a>
					</li>
					<li class="current tsd-kind-module">
						<a href="../modules/_src_programs_fsp_api_enum_intersolve_result_code_enum_.html">&quot;src/programs/fsp/api/enum/intersolve-<wbr>result-<wbr>code.enum&quot;</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
				<ul class="current">
					<li class="current tsd-kind-enum tsd-parent-kind-module">
						<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html" class="tsd-kind-icon">Intersolve<wbr>Result<wbr>Code</a>
						<ul>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#authorizationfailed" class="tsd-kind-icon">Authorization<wbr>Failed</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cannotcancel" class="tsd-kind-icon">Cannot<wbr>Cancel</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardblocked" class="tsd-kind-icon">Card<wbr>Blocked</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardexpired" class="tsd-kind-icon">Card<wbr>Expired</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardinactive" class="tsd-kind-icon">Card<wbr>Inactive</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardorcustomerdoesnotexist" class="tsd-kind-icon">Card<wbr>OrCustomer<wbr>Does<wbr>Not<wbr>Exist</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#cardunavailable" class="tsd-kind-icon">Card<wbr>Unavailable</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#eaninvalid" class="tsd-kind-icon">EANInvalid</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#incorrectactivationorpurchasevalue" class="tsd-kind-icon">Incorrect<wbr>Activation<wbr>OrPurchase<wbr>Value</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#invalidorunknownretailer" class="tsd-kind-icon">Invalid<wbr>OrUnknown<wbr>Retailer</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#invalidparameters" class="tsd-kind-icon">Invalid<wbr>Parameters</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#ok" class="tsd-kind-icon">Ok</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#pininvalid" class="tsd-kind-icon">PINInvalid</a>
							</li>
							<li class=" tsd-kind-enum-member tsd-parent-kind-enum">
								<a href="_src_programs_fsp_api_enum_intersolve_result_code_enum_.intersolveresultcode.html#unabletocancel" class="tsd-kind-icon">Unable<wbr>ToCancel</a>
							</li>
						</ul>
					</li>
				</ul>
				<ul class="after-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-object-literal"><span class="tsd-kind-icon">Object literal</span></li>
				<li class="tsd-kind-variable"><span class="tsd-kind-icon">Variable</span></li>
				<li class="tsd-kind-function"><span class="tsd-kind-icon">Function</span></li>
				<li class="tsd-kind-type-alias"><span class="tsd-kind-icon">Type alias</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-interface"><span class="tsd-kind-icon">Interface</span></li>
				<li class="tsd-kind-interface tsd-has-type-parameter"><span class="tsd-kind-icon">Interface with type parameter</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-class"><span class="tsd-kind-icon">Class</span></li>
				<li class="tsd-kind-class tsd-has-type-parameter"><span class="tsd-kind-icon">Class with type parameter</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-enum"><span class="tsd-kind-icon">Enumeration</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>