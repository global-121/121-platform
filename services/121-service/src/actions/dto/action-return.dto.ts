import {
  ActionArray,
  ActionType,
} from '@121-service/src/actions/action.entity';
import { UserOwnerDto } from '@121-service/src/user/dto/user-owner.dto';
import { ApiProperty } from '@nestjs/swagger';

export class ActionReturnDto {
  @ApiProperty({ example: 1, type: 'number' })
  id: number;
  @ApiProperty({ example: 1, enum: ActionArray })
  actionType: ActionType;
  @ApiProperty({ type: () => UserOwnerDto })
  user: UserOwnerDto;
  @ApiProperty({ example: '2021-09-01T00:00:00.000Z' })
  created: Date;
}
