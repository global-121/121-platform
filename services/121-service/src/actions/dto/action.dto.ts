import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsNumber, IsIn } from 'class-validator';
import { ActionType, ActionArray } from '../action.entity';

export class ActionDto {
  @ApiProperty({
    enum: ActionArray,
    example: ActionArray.join(' | '),
  })
  @IsNotEmpty()
  @IsIn(ActionArray)
  public readonly actionType: ActionType;
}
