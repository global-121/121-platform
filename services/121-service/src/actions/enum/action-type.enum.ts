export enum ActionType {
  startBlockNewPayment = 'start-block-new-payment',
  endBlockNewPayment = 'end-block-new-payment',
}
