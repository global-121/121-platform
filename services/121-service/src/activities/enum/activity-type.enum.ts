export enum ActivityTypeEnum {
  DataChange = 'data-change',
  FinancialServiceProviderChange = 'financial-service-provider-change',
  Message = 'message',
  Note = 'note',
  StatusChange = 'status-change',
  Transaction = 'transaction',
}
