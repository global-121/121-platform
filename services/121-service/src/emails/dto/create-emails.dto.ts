export interface CreateUserEmailPayload {
  email: string;
  displayName: string;
  password?: string;
}

export interface GenericEmailPayload {
  email: string;
  subject: string;
  body: string;
}
