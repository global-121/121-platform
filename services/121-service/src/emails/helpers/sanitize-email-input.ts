import { stripHtmlTags } from '@121-service/src/utils/strip-html-tags.helper';

export function sanitizeEmailInput<T extends { displayName: string }>(
  input: T,
): T {
  return {
    ...input,
    displayName: stripHtmlTags(input.displayName),
  };
}
