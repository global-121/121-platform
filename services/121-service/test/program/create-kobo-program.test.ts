/* eslint-disable jest/no-conditional-expect */

import { HttpStatus } from '@nestjs/common';

import { FinancialServiceProviders } from '@121-service/src/financial-service-providers/enum/financial-service-provider-name.enum';
import { getFinancialServiceProviderSettingByNameOrThrow } from '@121-service/src/financial-service-providers/financial-service-provider-settings.helpers';
import { CreateProgramFinancialServiceProviderConfigurationDto } from '@121-service/src/program-financial-service-provider-configurations/dtos/create-program-financial-service-provider-configuration.dto';
import { SeedScript } from '@121-service/src/scripts/enum/seed-script.enum';
import { LanguageEnum } from '@121-service/src/shared/enum/language.enums';
import {
  getKoboIntegration,
  getProgram,
  linkKoboForm,
  postProgram,
} from '@121-service/test/helpers/program.helper';
import { postProgramFinancialServiceProviderConfiguration } from '@121-service/test/helpers/program-financial-service-provider-configuration.helper';
import {
  getAccessToken,
  resetDB,
} from '@121-service/test/helpers/utility.helper';

describe('Create program which should be edited via kobo later', () => {
  let accessToken: string;

  beforeEach(async () => {
    await resetDB(SeedScript.productionInitialState);
    accessToken = await getAccessToken();
  });

  it('should go through program creation flow', async () => {
    // Arrange
    const program = {
      titlePortal: {
        en: 'Kobo land program',
      },
      currency: 'MWK',
      languages: [LanguageEnum.en, LanguageEnum.nl],
      fixedTransferValue: 20,
    };

    // Act
    const createProgramResponse = await postProgram(program, accessToken);
    expect(createProgramResponse.statusCode).toBe(HttpStatus.CREATED);
    console.log('ðŸš€ ~ it ~ createProgramResponse:', createProgramResponse.body);

    // Assert
    const programId = createProgramResponse.body.id;
    const getProgramResponse = await getProgram(programId, accessToken);
    console.log('ðŸš€ ~ it ~ getProgramResponse:', getProgramResponse.body);
    expect(createProgramResponse.statusCode).toBe(HttpStatus.CREATED);

    // Check for presence of the automatically created attributes
    const { programRegistrationAttributes } = getProgramResponse.body;
    expect(programRegistrationAttributes).toBeDefined();
    expect(Array.isArray(programRegistrationAttributes)).toBe(true);

    // Check for phoneNumber attribute
    const phoneNumberAttribute = programRegistrationAttributes.find(
      (attr) => attr.name === 'phoneNumber',
    );
    expect(phoneNumberAttribute).toBeDefined();
    expect(phoneNumberAttribute.type).toBe('text');

    // Check for fullName attribute
    const fullNameAttribute = programRegistrationAttributes.find(
      (attr) => attr.name === 'fullName',
    );
    expect(fullNameAttribute).toBeDefined();
    expect(fullNameAttribute.type).toBe('text');
    expect(fullNameAttribute.isRequired).toBe(true);

    const createProgramFspConfigurationDto: CreateProgramFinancialServiceProviderConfigurationDto =
      {
        name: 'Safaricom',
        label: {
          en: 'Safaricom label English translation',
          nl: 'Safaricom label Dutch translation',
        },
        isDefault: true,
        financialServiceProviderName: FinancialServiceProviders.safaricom,
        properties: [],
      };

    const postFspConfigResponse =
      await postProgramFinancialServiceProviderConfiguration({
        programId,
        body: createProgramFspConfigurationDto,
        accessToken,
      });

    console.log('ðŸš€ ~ it ~ postFspConfigResponse:', postFspConfigResponse.body);

    // Ensure that missing fsp attributes are added to the program
    const fsp = getFinancialServiceProviderSettingByNameOrThrow(
      createProgramFspConfigurationDto.financialServiceProviderName,
    );
    const fspAttributes = fsp.attributes.map((attribute) => attribute.name);
    const programWithFsp = await getProgram(programId, accessToken);
    const currentProgramAttributeNames =
      programWithFsp.body.programRegistrationAttributes.map(
        (attribute) => attribute.name,
      );
    expect(currentProgramAttributeNames).toEqual(
      expect.arrayContaining(fspAttributes),
    );

    // Test Kobo integration
    // Link Kobo form to program
    const koboLinkDto = {
      koboToken: process.env.KOBO_TOKEN,
      koboAssetId: process.env.KOBO_ASSET_ID,
      koboUrl: process.env.KOBO_URL,
    };
    console.log('ðŸš€ ~ it ~ koboLinkDto:', koboLinkDto);

    const linkKoboResponse = await linkKoboForm(
      programId,
      koboLinkDto,
      accessToken,
    );
    console.log('ðŸš€ ~ it ~ linkKoboResponse:', linkKoboResponse.body);

    expect(linkKoboResponse.status).toBe(HttpStatus.OK);

    // Verify Kobo integration was created successfully
    const getKoboResponse = await getKoboIntegration(programId, accessToken);

    expect(getKoboResponse.status).toBe(HttpStatus.OK);
    expect(getKoboResponse.body).toHaveProperty(
      'assetId',
      koboLinkDto.koboAssetId,
    );
    expect(getKoboResponse.body).toHaveProperty(
      'tokenCode',
      koboLinkDto.koboToken,
    );
    expect(getKoboResponse.body).toHaveProperty('programId', programId);
    expect(getKoboResponse.body).toHaveProperty('versionId');
    expect(getKoboResponse.body).toHaveProperty('id');
  });
});
