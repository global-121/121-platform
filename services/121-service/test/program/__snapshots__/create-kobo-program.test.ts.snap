// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create program which should be edited via kobo later should not allow linking of kobo form when expected survey items are missing 1`] = `
[
  "FSP "Safaricom" required attributes that are missing from the Kobo form: phoneNumber, nationalId",
  "FSP "Intersolve-visa" required attributes that are missing from the Kobo form: fullName, addressCity, addressHouseNumberAddition, addressPostalCode, addressStreet, phoneNumber",
  "Kobo form attribute "addressHouseNumber" has incompatible type for FSP integration 'Intersolve-visa': expected one of these: "integer, decimal, range", got "text"",
  "Kobo form must contain a question with name phoneNumber (should be a text type and country code should be included)",
  "Kobo form must contain the following name attributes defined in program.fullnameNamingConvention. However the following attributes are missing: fullName",
  "Kobo form must not contain a matrix item. Found: ["This is matrix question, which we do not support :(",null,null]",
]
`;
